lib/literal/array.rb

```diff
@@ -2,18 +2,31 @@
 
 class Literal::Array
 	class Generic
+		include Literal::Type
+
 		def initialize(type)
 			@type = type
 		end
 
+		attr_reader :type
+
 		def new(*value)
 			Literal::Array.new(value, type: @type)
 		end
 
 		alias_method :[], :new
 
 		def ===(value)
-			Literal::Array === value && @type == value.__type__
+			Literal::Array === value && Literal.subtype?(value.__type__, of: @type)
+		end
+
+		def <=>(other)
+			case other
+			when Literal::Array::Generic
+				@type <=> other.type
+			else
+				-1
```

Comment:
This should probably fall back to some `super` implementation on `Literal::Type` which could at least include `_Any?`, `_Never`, etc.